lib/mixins/http.js

```diff
@@ -20,10 +20,12 @@ const mixin = (superclass) => class extends superclass {
     return properties;
   }
 
-  static fetchAll(options) {
-    const url = this.buildUrl();
+  static fetchAll({ url, config = {} } = {}) {
+    url = url || this.buildUrl();
+    const clonedConfig = { ...config };
+    clonedConfig.params = this.encode(clonedConfig.params);
```

Comment:
don't we want a deep clone ?

Reply:
encode method should not modify the object. if it does, it's the dev problem.